packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -221,10 +221,9 @@ class CupertinoSheetTransition extends StatefulWidget {
       reverseCurve: reverseCurve,
       parent: secondaryAnimation,
     );
-    final double deviceCornerRadius = MediaQuery.maybeViewPaddingOf(context)?.top ?? 0;
 
     final Animatable<BorderRadiusGeometry> decorationTween = Tween<BorderRadiusGeometry>(
-      begin: BorderRadius.circular(deviceCornerRadius),
+      begin: BorderRadius.circular(0),
       end: BorderRadius.circular(12),
     );
 
```

Comment:
Below this, I think it's a good idea to change the radius transition to this:
``` dart
        AnimatedBuilder(
          animation: radiusAnimation,
          child: child,
          builder: (BuildContext context, Widget? child) {
            return ClipRRect(
              borderRadius:
                  !secondaryAnimation.isDismissed
                      ? radiusAnimation.value
                      : BorderRadius.circular(0),
              child: contrastedChild,
            );
          },
        ),
```
Switching back to zero here ensures the rounded corner does not stick around once the animation is finished. Without the `0.9` smoothing I was finding this caused the transition to jump on some screens. But that was a sign it wasn't finishing correctly.